Demo/Common/IntegrationExamples/Components/ApplePayComponentAdvancedFlowExample.swift

```diff
@@ -0,0 +1,240 @@
+        return component
+    }
+
+    // MARK: - Payment response handling
+
+    private func paymentResponseHandler(result: Result<PaymentsResponse, Error>) {
+        switch result {
+        case let .success(response):
+            if let action = response.action {
+                adyenActionComponent.handle(action)
+            } else {
+                finish(with: response)
+            }
+        case let .failure(error):
+            finish(with: error)
+        }
+    }
+
+    internal func finish(with result: PaymentsResponse) {
+        let success = result.resultCode == .authorised || result.resultCode == .received || result.resultCode == .pending
+        let message = "\(result.resultCode.rawValue) \(result.amount?.formatted ?? "")"
+        finalize(success, message)
+    }
+
+    internal func finish(with error: Error) {
+        let message: String
+        if let componentError = (error as? ComponentError), componentError == ComponentError.cancelled {
+            message = "Cancelled"
+        } else {
+            message = error.localizedDescription
+        }
+        finalize(false, message)
+    }
+
+    private func finalize(_ success: Bool, _ message: String) {
+        applePayComponent?.finalizeIfNeeded(with: success) { [weak self] in
+            guard let self = self else { return }
+            self.dismissAndShowAlert(success, message)
+        }
+    }
+
+    internal func dismissAndShowAlert(_ success: Bool, _ message: String) {
+        presenter?.dismiss {
+            // Payment is processed. Add your code here.
+            let title = success ? "Success" : "Error"
+            self.presenter?.presentAlert(withTitle: title, message: message)
+        }
+    }
+
+    // MARK: - Presentation
+
+    private func present(_ component: PresentableComponent) {
+        guard component.requiresModalPresentation else {
+            presenter?.present(viewController: component.viewController, completion: nil)
+            return
+        }
+
+        let navigation = UINavigationController(rootViewController: component.viewController)
+        component.viewController.navigationItem.rightBarButtonItem = .init(barButtonSystemItem: .cancel,
+                                                                           target: self,
+                                                                           action: #selector(cancelPressed))
+        presenter?.present(viewController: navigation, completion: nil)
+    }
+
+    @objc private func cancelPressed() {
+        applePayComponent?.cancelIfNeeded()
+        presenter?.dismiss(completion: nil)
+    }
+
+}
+
+extension ApplePayComponentAdvancedFlowExample: PaymentComponentDelegate {
+
+    internal func didSubmit(_ data: PaymentComponentData, from component: PaymentComponent) {
+        let request = PaymentsRequest(data: data)
+        apiClient.perform(request) { [weak self] result in
+            self?.paymentResponseHandler(result: result)
```

Comment:
There is no Action, so we can resolve `result` directly:
```suggestion
        switch result {
        case let .success(response):
            finish(with: response)
        case let .failure(error):
            finish(with: error)
        }
```

Reply:
Resolved. 47a1fb69f153def70200c018673e639d6796d548